mediatek: filogic: Zyxel EX5601-T0 dts fix and cleanup

testuser7 opened the pull request:
1. Fix incorrect "fip" partition size.
2. Dts cleanup for Zyxel EX5601:
- duplicated code
- trailing zeros and whitespaces
3. Fix 5G MAC address
4. Add label wan and cpu

testuser7 committed:
mediatek: filogic: fix fip partition size for Zyxel EX5601-T0

Fix incorrect "fip" partition size.

Signed-off-by: Paweł Owoc <frut3k7@gmail.com>
Link: https://github.com/openwrt/openwrt/pull/19968
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

testuser7 committed:
mediatek: filogic: Zyxel EX5601-T0 dts cleanup

Dts cleanup for Zyxel EX5601:
- duplicated code
- trailing zeros and whitespaces

Signed-off-by: Paweł Owoc <frut3k7@gmail.com>
Link: https://github.com/openwrt/openwrt/pull/19968
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

castiel652 commented:
You should define precal as well

castiel652 commented:
Pre-calibration is in Factory partition.
If you use NVMEM to read EEPROM then you will need to explicitly use it for pre-cal as well.
For reference https://github.com/openwrt/openwrt/pull/14412, but change the reg to `reg = <0x1010 0x6810>`

testuser7 commented:
> Pre-calibration is in Factory partition.

It is defined:
- https://github.com/openwrt/openwrt/pull/19968/files#diff-1089a1ab2879f3039e67226aaf5af421df5e85c4e79b6b682b3f2318075cd01fL52-L55
- https://github.com/openwrt/openwrt/pull/19968/files#diff-4590e5985b4f1dc2ddd77726f8cbfc949aa05809ab2b676f92fc115a3897333cL51-L54

castiel652 commented:
Ah in that case, then you can ignore my comment.

testuser7 committed:
mediatek: filogic: fix 5G MAC address for Zyxel EX5601

Currently 5G wireless interface MAC address is incorrect.
Fix by setting MAC address using Factory data.

Signed-off-by: Paweł Owoc <frut3k7@gmail.com>
Link: https://github.com/openwrt/openwrt/pull/19968
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

testuser7 committed:
mediatek: filogic: Add label wan and cpu for Zyxel EX5601-T0

Add labels wan and cpu for ethernet ports.

Signed-off-by: Paweł Owoc <frut3k7@gmail.com>
Link: https://github.com/openwrt/openwrt/pull/19968
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

Gingernut1978 commented:
@testuser7 

Still needed after these commits?

11_fix_wifi_mac
```
	zyxel,ex5601-t0-stock|\
	zyxel,ex5601-t0-ubootmod)
		addr=$(mtd_get_mac_binary "Factory" 0x4)
		[ "$PHYNBR" = "1" ] && macaddr_add $addr 1 > /sys${DEVPATH}/macaddress
```

testuser7 commented on target/linux/mediatek/dts/mt7986a-zyxel-ex5601-t0-common.dtsi:
```diff
@@ -285,6 +286,7 @@
 
 				port@6 {
 					reg = <6>;
+					label = "cpu";
```
It is not needed but helps identify the port connected to the SoC.

testuser7 replied:
I removed it as this information is not visible in the system anyway.

hrw commented:
Next time please take care of upgrade path.

Had to edit WAN interfaces as `eth1` disappeared when I upgraded to 25.12: #21331
